src/main/java/com/cloudogu/scm/mirror/api/MirrorRootResource.java

```diff
@@ -72,9 +70,9 @@ public MirrorRootResource(Provider<MirrorResource> mirrorResource, RepositoryLin
   @POST
   @Path("/repositories")
   @Consumes("application/json")
-  public Response createMirrorRepository(@Valid MirrorRequestDto requestDto) {
-    Repository repository = requestDtoToRepositoryMapper.map(requestDto);
-    MirrorConfiguration configuration = requestDtoToRequestMapper.map(requestDto);
+  public Response createMirrorRepository(@Valid MirrorCreationDto requestDto) {
+    Repository repository = mirrorCreationDtoMapper.mapToRepository(requestDto);
```

Comment:
To validate the (potentially added) certificate I'd add something like this:
```java
    doThrow().violation("certificate must not be empty")
      .when(
        requestDto.getCertificateCredential() != null && Strings.isNullOrEmpty(requestDto.getCertificateCredential().getCertificate())
      );
```

Reply:
Good catch, will add